spinalcordtoolbox/types.py

```diff
@@ -134,23 +130,6 @@ def __truediv__(self, scalar):
         return Coordinate([self.x / float(scalar), self.y / float(scalar), self.z / float(scalar), self.value])
 
 
-class CoordinateValue(Coordinate):
-    def __init__(self, coord=None, mode='index'):
```

Comment:
That's a fair summary! Some extra notes:

- I wanted to get rid of this class not just because it was barely used, but because it behaved so differently from its parent class (which breaks the Liskov substitution principle, etc.):
  - Equality checking for `Coordinate` objects compares the `x`, `y`, `z` coordinates, and ignores the optional `value` field entirely.
  - Equality checking for `CoordinateValue` objects do the exact opposite: ignore the `x`, `y`, `z` coordinates, and only compare the `value` field.
- I don't think the `__hash__` method was ever used, but this is a subtle point: Python uses the `__hash__` method under the hood when objects get inserted or looked up in sets, dicts, and a few other data structures. As the [docs for `__hash__` say](https://docs.python.org/3/reference/datamodel.html#object.__hash__):
  > The only required property is that objects which compare equal have the same hash value;

  So, for any classes which redefine `__eq__`, it's important to redefine `__hash__` as well, otherwise their instances will behave _very_ strangely with sets and dicts.